staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -438,15 +478,31 @@ func (p *applyPatcher) applyPatchToCurrentObject(obj runtime.Object) (runtime.Ob
 		return nil, errors.NewBadRequest(fmt.Sprintf("error decoding YAML: %v", err))
 	}
 
-	return p.fieldManager.Apply(obj, patchObj, p.options.FieldManager, force)
+	if p.validationDirective == metav1.FieldValidationStrict {
+		strictPatchObj := &unstructured.Unstructured{Object: map[string]interface{}{}}
```

Comment:
is there a way to make use of the decoded object rather than decoding multiple times?

Reply:
yeah, sounds like that's what we need, make sure that's tracked somewhere

Reply:
A little late, but currently tracking at https://github.com/kubernetes-sigs/yaml/issues/70